x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/filestructurefinder/FieldStats.java

```diff
@@ -142,6 +142,15 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static Object toIntegerIfInteger(double d) {
```

Comment:
I am not sure how strict we are with typing, but since this always returns a boxed `double` or `int` it would make sense for it to be a `Number` instead of just an `Object` right?